src/Shared/PackageManagers/BaseNuGetProjectManager.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.PackageManagers
+{
+    using Contracts;
+    using PackageUrl;
+    using Model.Metadata;
+    using PackageActions;
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+
+    public abstract class BaseNuGetProjectManager : TypedManager<NuGetPackageVersionMetadata, BaseNuGetProjectManager.NuGetArtifactType>
+    {
+        /// <summary>
+        /// The type of the project manager from the package-url type specifications.
+        /// </summary>
+        /// <seealso href="https://www.github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst"/>
+        public const string Type = "nuget";
+
+        public override string ManagerType => Type;
+
+        protected BaseNuGetProjectManager(IManagerPackageActions<NuGetPackageVersionMetadata> actions, IHttpClientFactory httpClientFactory, string directory, TimeSpan? timeout = null) : base(actions, httpClientFactory, directory, timeout)
+        {
+        }
+
+        public enum NuGetArtifactType
+        {
+            Unknown = 0,
+            Nupkg,
+            Nuspec,
+        }
+
+        internal static BaseNuGetProjectManager Create(string destinationDirectory, IHttpClientFactory httpClientFactory, TimeSpan? timeout, PackageURL? packageUrl)
+        {
+            if (packageUrl is null || packageUrl.Qualifiers?.TryGetValue("repository_url", out string? repositoryUrlQualifier) != true ||
+                repositoryUrlQualifier?.TrimEnd('/') == NuGetProjectManager.NUGET_DEFAULT_INDEX)
+            {
+                return new NuGetProjectManager(destinationDirectory, NuGetPackageActions.CreateV3(), httpClientFactory, timeout);
+            }
+            else if (repositoryUrlQualifier?.TrimEnd('/') == NuGetV2ProjectManager.POWER_SHELL_GALLERY_DEFAULT_INDEX)
+            {
+                return new NuGetV2ProjectManager(destinationDirectory, NuGetPackageActions.CreateV2(), httpClientFactory, timeout);
+            }
+            else
+            {
+                throw new NotImplementedException($"NuGet package URLs having a repository URL other than '{NuGetV2ProjectManager.POWER_SHELL_GALLERY_DEFAULT_INDEX}' or '{NuGetProjectManager.NUGET_DEFAULT_INDEX}' are not currently supported.");
+            }
+        }
+
+        /// <inheritdoc />
+        public override async IAsyncEnumerable<PackageURL> GetPackagesFromOwnerAsync(string owner, bool useCache = true)
+        {
+            throw new NotImplementedException("Haven't found a way to implement this yet!");
```

Comment:
Was this already unimplemented on the existing NuGetProjectManager? Or is this a TODO to be fixed during this PR?